base_layer/core/src/proof_of_work/lwma_diff.rs

```diff
@@ -17,54 +17,60 @@ pub const LOG_TARGET: &str = "c::pow::lwma_diff";
 
 pub struct LinearWeightedMovingAverage {
     timestamps: VecDeque<EpochTime>,
-    accumulated_difficulties: VecDeque<Difficulty>,
+    target_difficulties: VecDeque<Difficulty>,
     block_window: usize,
     target_time: u64,
-    initial_difficulty: u64,
+    initial_difficulty: Difficulty,
+    max_block_time: u64,
 }
 
 impl LinearWeightedMovingAverage {
-    pub fn new(block_window: usize, target_time: u64, initial_difficulty: u64) -> LinearWeightedMovingAverage {
+    pub fn new(
+        block_window: usize,
+        target_time: u64,
+        initial_difficulty: Difficulty,
+        max_block_time: u64,
+    ) -> LinearWeightedMovingAverage
+    {
         LinearWeightedMovingAverage {
             timestamps: VecDeque::with_capacity(block_window + 1),
-            accumulated_difficulties: VecDeque::with_capacity(block_window + 1),
+            target_difficulties: VecDeque::with_capacity(block_window + 1),
             block_window,
             target_time,
             initial_difficulty,
+            max_block_time,
         }
     }
 
     fn calculate(&self) -> Difficulty {
         let timestamps = &self.timestamps;
         if timestamps.len() <= 1 {
             // return INITIAL_DIFFICULTY;
-            return self.initial_difficulty.into();
+            return self.initial_difficulty;
         }
 
         // Use the array length rather than block_window to include early cases where the no. of pts < block_window
         let n = (timestamps.len() - 1) as u64;
 
         let mut weighted_times: u64 = 0;
-
-        let difficulty: u64 = self.accumulated_difficulties[n as usize]
-            .checked_sub(self.accumulated_difficulties[0])
-            .expect("Accumulated difficulties cannot decrease in proof of work")
-            .as_u64();
-        let ave_difficulty = difficulty as f64 / n as f64;
+        let mut difficulty: u64 = 0;
+        for diff in self.target_difficulties.iter() {
```

Comment:
use a for loop. You need to ignore the first difficulty. 
You are calculating the avg between 151 difficulties and not 150.

Reply:
Good find. This has been updated

Reply:
Don't use a for loop. `skip(1)`. 
Edit: That's what you did.  ;)